packages/vite/src/node/config.ts

```diff
@@ -538,21 +542,31 @@ export async function resolveConfig(
   const createResolver: ResolvedConfig['createResolver'] = (options) => {
     let aliasContainer: PluginContainer | undefined
     let resolverContainer: PluginContainer | undefined
+
+    const plugins =
+      options?.plugins?.filter(
+        (p) => p.resolveId && p.name !== 'alias' && !p.name.startsWith('vite:')
+      ) || []
+
     return async (id, importer, aliasOnly, ssr) => {
       let container: PluginContainer
       if (aliasOnly) {
         container =
           aliasContainer ||
           (aliasContainer = await createPluginContainer({
             ...resolved,
-            plugins: [aliasPlugin({ entries: resolved.resolve.alias })]
+            plugins: [
+              ...plugins,
+              aliasPlugin({ entries: resolved.resolve.alias })
+            ]
           }))
       } else {
         container =
           resolverContainer ||
           (resolverContainer = await createPluginContainer({
             ...resolved,
             plugins: [
+              ...plugins,
```

Comment:
Here it is tricky to properly order the plugins. The resolvePlugin should be injected in the same place where the regular resolve plugin is placed, no? Maybe the logic at 546 should generate two arrays, one for plugins before resolve, and one for the ones after it.